src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+			supportsRed: response.supportsRed!,
+			supportsGreen: response.supportsGreen!,
+			supportsBlue: response.supportsBlue!,
+			supportsAmber: response.supportsAmber!,
+			supportsCyan: response.supportsCyan!,
+			supportsPurple: response.supportsPurple!,
+			supportsIndex: response.supportsIndex!,
+		};
+	}
+
+	public async get(colorKey: ColorKey): Promise<ColorSwitchGetResult> {
+		this.assertSupportsCommand(ColorSwitchCommand, ColorSwitchCommand.Get);
+
+		const colorComponent = getColorComponentFromKey(colorKey);
+		if (colorComponent == undefined) {
+			throw new Error(`Unsupported color "${colorKey}".`);
+		}
+
+		const cc = new ColorSwitchCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent: colorComponent,
+		});
+		const response = (await this.driver.sendCommand<ColorSwitchCCReport>(
+			cc,
+		))!;
+		return {
+			colorComponent: response.colorComponent,
+			currentValue: response.currentValue,
+			targetValue: response.targetValue,
+			duration: response.duration,
+		};
+	}
+
+	public async set(colorTable: ColorTable): Promise<void> {
+		this.assertSupportsCommand(ColorSwitchCommand, ColorSwitchCommand.Set);
+
+		const cc = new ColorSwitchCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			...colorTable,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	public async startLevelChange(
+		colorKey: ColorKey,
+		direction: StartLevelChangeDirection,
+		options: StartLevelChangeOptions = {},
+	): Promise<void> {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.StartLevelChange,
+		);
+
+		const colorComponent = getColorComponentFromKey(colorKey);
+		if (colorComponent == undefined) {
+			throw new Error(`Unknown color "${colorKey}".`);
+		}
+
+		const cc = new ColorSwitchCCStartLevelChange(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent,
+			down: direction === "down",
+			startLevel: options.startLevel,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	public async stopLevelChange(colorKey: ColorKey): Promise<void> {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.StopLevelChange,
+		);
+
+		const colorComponent = getColorComponentFromKey(colorKey);
+		if (colorComponent == undefined) {
+			throw new Error(`Unknown color "${colorKey}".`);
+		}
+
+		const cc = new ColorSwitchCCStopLevelChange(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	) => {
+		if (property !== "targetColor") {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		}
+
+		if (!propertyKey) {
+			// No error throw util for this?
```

Comment:
Yep, I've stumbled across this aswell. Should probably be merged with `throwUnsupportedProperty`